bindings/wasm/examples/src/1_advanced/3_did_issues_tokens.ts

```diff
@@ -0,0 +1,181 @@
+    // Initially, all carbon credits are owned by the foundry.
+    nativeTokens: [
+      {
+        id: tokenId,
+        amount: HexHelper.fromBigInt256(bigInt(500_000))
+      }
+    ],
+    // The authority is set as the immutable owner.
+    unlockConditions: [
+      {
+        type: IMMUTABLE_ALIAS_UNLOCK_CONDITION_TYPE,
+        address: {
+          type: ALIAS_ADDRESS_TYPE,
+          aliasId: authorityDid.toAliasId()
+        }
+      }
+    ]
+  });
+
+  // Set the appropriate storage deposit.
+  carbonCreditsFoundry.amount = TransactionHelper.getStorageDeposit(carbonCreditsFoundry, rentStructure).toString();
+
+  // Publish the foundry.
+  const [blockId, block] = await client.buildAndPostBlock(secretManager, { outputs: [authorityAliasOutput, carbonCreditsFoundry] });
+  await client.retryUntilIncluded(blockId);
+
+  // ===================================
+  // Resolve foundry and its issuer DID.
+  // ===================================
+
+  // Get the latest output that contains the foundry.
+  const carbonCreditsFoundryId: string = tokenId;
+  const outputId: string = await client.foundryOutputId(carbonCreditsFoundryId)
+  const outputResponse: IOutputResponse = await client.getOutput(outputId);
+  const output: OutputTypes = outputResponse.output;
+
+  if (output.type === FOUNDRY_OUTPUT_TYPE) {
+    carbonCreditsFoundry = output;
+  } else {
+    throw new Error("expected foundry output")
+  }
+
+  // Get the Alias Id of the authority that issued the carbon credits foundry.
+  // Non-null assertion is safe as each founry output needs to have an immutable alias unlock condition.
+  const aliasUnlockCondition: IImmutableAliasUnlockCondition = carbonCreditsFoundry.unlockConditions.find(unlockCondition => unlockCondition.type === IMMUTABLE_ALIAS_UNLOCK_CONDITION_TYPE)! as IImmutableAliasUnlockCondition;
+
+  // We know the immutable alias unlock condition contains an alias address.
+  const authorityAliasId: string = (aliasUnlockCondition.address as IAliasAddress).aliasId
+
+  // Reconstruct the DID of the authority.
+  authorityDid = new IotaDID(Converter.hexToBytes(authorityAliasId), networkName);
+
+  // Resolve the authority's DID document.
+  authorityDocument = await didClient.resolveDid(authorityDid);
+
+  console.log("The authority's DID is:", JSON.stringify(authorityDocument, null, 2));
+
+  // =========================================================
+  // Transfer 1000 carbon credits to the address of a company.
+  // =========================================================
+
+  // Create a new address that represents the company.
+  const companyAddressBech32: string = (await client.generateAddresses(secretManager, {
+    accountIndex: 0,
+    range: {
+      start: 1,
+      end: 2,
+    },
+  }))[0];
+  const companyAddress = Bech32Helper.addressFromBech32(companyAddressBech32, networkName);
+
+  // Create a timestamp 24 hours from now.
+  const tomorrow: number = Math.floor(Date.now() / 1000) + (60 * 60 * 24);
```

Comment:
Could we use the library's `Timestamp` and `Duration` to simplify this? The only problem was the lack of `.toUnix` due to u64/bigint problems in wasm-bindgen`, but that seems to have been fixed with the latest release? https://github.com/rustwasm/wasm-bindgen/pull/3037

Maybe an issue is needed, if only to expose `.toUnix`.

Reply:
Yeah, I had it with `Timestamp` and `Duration` originally but indeed the lack of `toUinx` was the reason to use `Date.now` instead. Would be nice if we could revert this PR completely: https://github.com/iotaledger/identity.rs/pull/365. 
The reason for removal were issues on React Native specifically (#362), is that no longer a problem then?